ddtrace/contrib/pytest/plugin.py

```diff
@@ -30,6 +32,17 @@ def _store_span(item, span):
     setattr(item, "_datadog_span", span)
 
 
+def _json_encode(params):
+    # type: (Dict[str, Any]) -> str
+    """JSON encode parameters. If complex object show inner values, otherwise default to string representation."""
+
+    def inner_encode(obj):
+        obj_dict = getattr(obj, "__dict__", None)
+        return obj_dict if obj_dict else str(obj)
```

Comment:
can `str(obj)` fail?

I think `repr(obj)` would make more sense here, it shouldn't fail unless there is a `__repr__` defined that has side effects

Reply:
`str(obj)` could be a source of unwanted side-effects (think lazy object)